[Chantell Yu] iP

chanyuk opened the pull request:
# CHAN BOT

> super great! [source](https://github.com/nus-cs2103-AY2223S1/ip)

CHAN BOT frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

* [X] Managing tasks
* [ ] Managing deadlines
* [ ] Reminders (coming soon)

If you're a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

chanyuk committed:
Add Greet, Echo, Exit

chanyuk committed:
Add Add, List

chanyuk committed:
Add Mark as Done

chanyuk committed:
Add ToDos, Events, Deadlines

chanyuk committed:
Add Automated Text UI Testing and Handle Errors

chanyuk committed:
Add Delete

chanyuk committed:
Add packages tasks and dukeexceptions

chanyuk committed:
Add more OOP and abstract out Commands

chanyuk committed:
Add Save

chanyuk committed:
Add Dates and Times

chanyuk committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/tasks/Deadlines.java
#	src/main/java/tasks/Events.java

chanyuk committed:
Add more OOP

chanyuk committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

chanyuk committed:
Add JUnit Tests

chanyuk committed:
JavaDoc

chanyuk committed:
Follow the Coding Standard

chanyuk committed:
Add Find

chanyuk committed:
Add JavaDoc

chanyuk committed:
Follow the Coding Standard

chanyuk committed:
Add Find

chanyuk committed:
Merge branch 'branch-A-JavaDoc'

# Conflicts:
#	src/main/java/duke/Parser.java

chanyuk committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/commands/ByeCommand.java
#	src/main/java/duke/Parser.java

chanyuk committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/commands/FindCommand.java
#	src/main/java/duke/Parser.java

danzzzerl commented on src/main/java/commands/ListCommand.java:
```diff
@@ -0,0 +1,35 @@
+     */
+    public void run(TaskList taskList) {
+        if (taskList.getSize() == 0) {
+            System.out.println("You have no tasks at the moment!");
```
Not sure but maybe could put this under Ui? Since it is kind of a user interaction. Can then call Ui.noTasksMessage() or something like that.

danzzzerl commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,60 @@
+import tasks.TaskList;
+
+
+public class Duke {
```
Could add in JavaDoc for this class

Hongyi6328 replied:
Agreed. Perhaps more JavaDoc and header comments will increase readability?

danzzzerl commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,82 @@
+import dukeexceptions.NoSuchCommandException;
+
+public class Parser {
+    private enum Commands {
```
Find this use of enum very neat and tidy, nice!

danzzzerl commented on src/main/java/tasks/Deadlines.java:
```diff
@@ -0,0 +1,46 @@
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class Deadlines extends Task {
```
I think Deadline would be a better class name than Deadlines. For Events as well.

danzzzerl reviewed (commented):
Overall neat code, looks organised and easy to read. Great job!

Hongyi6328 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,72 @@
+        }
+    }
+
+    public void end(TaskList taskList) {
```
Maybe try a better and more meaningful name of this method? `end` seems a little bit confusing to me.

Hongyi6328 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,46 @@
+        System.out.println("What can I do for you?\n");
+    }
+
+    public static void printByeStatement() {
+        System.out.println("Bye bye!");
+        System.out.println(" /\\_/\\");
+        System.out.println("/ o o \\");
+        System.out.println("/    ^\\");
```
I like the personality here!

Hongyi6328 commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,49 @@
+package tasks;
+
+public class Task {
```
Would it be better if we can turn this class to an abstract class, because in real application, we don't actually instantiate a "general" task?

Hongyi6328 reviewed (commented):
Overall, very neat and impressive code. Nice job! I really love the personality of your chat bot. Keep up the good work. Some minor suggestions are given. Feel free to clarify with me if you have any question!

chanyuk committed:
Update README.md

chanyuk committed:
Update README.md

chanyuk committed:
Update README.md

chanyuk committed:
Update README.md

chanyuk committed:
Update README.md

chanyuk committed:
Update README.md

chanyuk committed:
Add GUI

chanyuk committed:
Merge commit '2c982213756a1af85e3ef7d848cb2d8a50e865eb'

chanyuk committed:
Add padding to chatbot

chanyuk committed:
Add assertions

Assertions define assumptions about the function calls so the runtime can verify them.

chanyuk committed:
Improve code quality

- Deleted unnecessary import statements
- Fixed typos in JavaDoc
- Restructured code to avoid lengthy methods
- Abstracted further to improve clarity

chanyuk committed:
Merge pull request #1 from chantellyu/branch-A-Assertions

Add assertions

chanyuk committed:
Merge pull request #2 from chantellyu/branch-A-CodeQuality

Improve code quality

chanyuk committed:
Merge branch 'branch-A-CodeQuality'

chanyuk committed:
Merge branch 'master' of https://github.com/chantellyu/ip

chanyuk committed:
Add C-Statistics

chanyuk committed:
Revert "Add C-Statistics"

This reverts commit d57f64f8f46971b4c5868f7944a87f8f50a46f2b.

chanyuk committed:
Add C-Statistics

The only way to track tasks done is to look at the entire tasklist via
the list command.

Providing a statistics command allows users to further break down their
tasklist.

Let's create a statistics command that shows users the tasks they have
completed and the tasks currently left incomplete.

chanyuk committed:
Fix JavaDoc comments

chanyuk committed:
Fix Storage issues

Storage does not store updated marked status of tasks if the command
mark or unmark is used.

Let's,
* allow todos, deadlines and events to have the ability to change the
completion status in the Task class.
*  change the string representation to reflect the accurate state of the
task.

chanyuk committed:
Add a unique personality

chanyuk committed:
Add Help Command, fix GUI

Chatbot does not display messages beyond 17 lines and cuts off text below.

Let's change font to Courier New to fix this.

chanyuk committed:
Add a User Guide

chanyuk committed:
Rename png example file to Ui.png

chanyuk committed:
Improve Ui, delete unneccessary code
